src-django/api/models.py

```diff
@@ -97,3 +97,70 @@ def save(self, **kwargs):
 
     class Meta:
         app_label = 'api'
+
+
+class Condition(models.Model):
+    CRITERIA_TYPES = (
+        ('EQUALS', 'EQUALS'),
+        ('GREATER', 'GREATER'),
+        ('LESSER', 'LESSER')
+    )
+
+    NODE_TYPES = (
+        ('AND', 'AND'),
+        ('OR', 'OR'),
+        ('NOT', 'NOT'),
+        ('CRITERIA', 'CRITERIA')
+    )
+
+    parent = models.ForeignKey('self', on_delete=models.CASCADE, related_name='children', blank=True, null=True)
+    element = models.ForeignKey(Element, blank=True, null=True)
+    node_type = models.CharField(max_length=8, choices=NODE_TYPES, null=True, blank=True)
+    criteria_type = models.CharField(max_length=8, choices=CRITERIA_TYPES, null=True, blank=True)
+    value = models.TextField(blank=True, null=True)
+    last_modified = models.DateTimeField(auto_now=True)
+    created = models.DateTimeField(auto_now_add=True)
+
+    def save(self, **kwargs):
+        if self.node_type:
+            if (self.node_type, self.node_type) not in self.NODE_TYPES:
+                raise IntegrityError('Invalid condition node type')
+
+            if self.node_type is 'CRITERIA':
+                if not self.criteria_type:
+                    raise IntegrityError('"CRITERIA" node type requires a criteria type')
+
+                if (self.criteria_type, self.criteria_type) not in self.CRITERIA_TYPES:
+                    raise IntegrityError('Invalid criteria_type for condition')
+
+                # if self.parent.count() is not 0:
+                #     raise IntegrityError('CRITERIA can not have child nodes')
+
+                if not self.element:
+                    raise IntegrityError('CRITERIA must have an element')
+
+            # if self.node_type is 'NOT' and len(self.children) is not 1:
+            #     raise IntegrityError('NOT must have exactly one child node')
+
+            # if self.children.count < 2:
+            #     raise IntegrityError('AND and OR nodes must have at least two children')
+
+        super(Condition, self).save()
+
+        if self.parent:
+            self.parent.last_modified = self.last_modified
+            self.parent.save()
+        elif self.show_if:
```

Comment:
It's referenced from the ShowIf model as a foreign key to a condition. There is no such thing as a OneToMany relationship, so we have to do it this way. It's kinda silly...